uc.c

```diff
@@ -773,6 +773,136 @@ static bool check_mem_area(uc_engine *uc, uint64_t address, size_t size)
     return (count == size);
 }
 
+uc_err uc_virtual_to_physical(uc_engine *uc, uint64_t address, uc_prot prot,
```

Comment:
Do we really need prot here? Is there possible that different protections lead to different physical addresses? If the protection is only used for access check, is it trivial to skip such checks and only "translate" addresses for whatever protection? If that is not trivial, we can keep it as it is.

Reply:
The prot is to have the mmu access rights checked. You might want to check if the corresponding address is write accessible. To check this you can call `uc_virtual_to_physical(uc, addr, UC_PROT_WRITE, &paddr)` and check the return code.

Also it's necessary to have prot for a potential `tlb_fill` call.

Reply:
But for “translation” itself, it has nothing to do with permissions, right? My point is that, from the semantics of the API, protection seems not necessary here though I understand tlb_fill requires this.

Reply:
For example, assume a memory region located at vaddr 0x2000 with paddr 0x1000 but with PROT_NONE, i.e. no one can access it. But from the perspective of emulator, the mapping is there exactly and we should be able to translate addresses no matter the protections.

However, I do understand that qemu code might not design for such cases. Therefore, if you think it is not feasible to translate addresses without protection, we can leave it as it is but with better docs.